crates/red_knot_python_semantic/src/types/builder.rs

```diff
@@ -56,7 +58,26 @@ impl<'db> UnionBuilder<'db> {
         self
     }
 
-    pub(crate) fn build(self) -> Type<'db> {
+    /// Performs the following normalizations:
+    ///     - Replaces `Literal[True,False]` with `bool`.
+    ///     - TODO For enums `E` with members `X1`,...,`Xn`, replaces
+    ///     `Literal[E.X1,...,E.Xn]` with `E`.
+    fn simplify(&mut self) {
+        if self
+            .elements
+            .is_superset(&[Type::BooleanLiteral(true), Type::BooleanLiteral(false)].into())
```

Comment:
nice approach!

Reply:
It would be interesting to look at the generated code. `into` creates an `OrderSet` which in turn allocates. 

I prefer using `self.elements.contains(Type::BooleanLiteral(true) && self.elements.contains(Type::BooleanLiteral(false))`, which does not depend on the compiler being able to see through that the allocation isn't necessary (which requires rather involved analysis)